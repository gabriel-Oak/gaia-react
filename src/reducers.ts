import { combineReducers } from 'redux';

import authReducer from './pages/auth/authReducer';
import mainReducer from './pages/main/mainReducer';

export interface ReducersPool {
  mainReducer: object;
  authReducer: object;
}

export default combineReducers({
  authReducer,
  mainReducer
});