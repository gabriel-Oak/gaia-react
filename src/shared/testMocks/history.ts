
export const historyMock: historyTesting = {
  goBack: () => {},
  push: () => {}
};

export const matchMock = {
  isExact: true,
  params: {},
  path: '/',
  url: '/hgfd'
};

export interface historyTesting {
  goBack: Function;
  push: Function;
}