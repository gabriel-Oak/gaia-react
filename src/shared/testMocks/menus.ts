export const menusMock = [
  {
    data: '2019-11-04T22:16:00.000Z',
    guarnicao1: 'Batata Frita',
    guarnicao2: 'Ameixa da jamaica',
    opcao1: 'Omelete',
    opcao2: 'Bife',
    pratoPrincipal: 'Bacon',
    salada: 'Alface e tomate',
    sobremesa: 'Maça',
    __v: 0,
    _id: '5d08c85f4d6ea30b43bf6d3b'
  },
  {
    data: '2019-11-05T22:16:00.000Z',
    guarnicao1: 'Batata Frita',
    guarnicao2: 'Ameixa da jamaica',
    opcao1: 'Omelete',
    opcao2: 'Bife',
    pratoPrincipal: 'Bacon',
    salada: 'Alface e tomate',
    sobremesa: 'Maça',
    __v: 0,
    _id: '5d08c85f4d6ea30b43bf6d3b'
  },
  {
    data: '2019-11-06T22:16:00.000Z',
    guarnicao1: 'Batata Frita',
    guarnicao2: 'Ameixa da jamaica',
    opcao1: 'Omelete',
    opcao2: 'Bife',
    pratoPrincipal: 'Bacon',
    salada: 'Alface e tomate',
    sobremesa: 'Maça',
    __v: 0,
    _id: '5d08c85f4d6ea30b43bf6d3b'
  },
  {
    data: '2019-11-07T22:16:00.000Z',
    guarnicao1: 'Batata Frita',
    guarnicao2: 'Ameixa da jamaica',
    opcao1: 'Omelete',
    opcao2: 'Bife',
    pratoPrincipal: 'Bacon',
    salada: 'Alface e tomate',
    sobremesa: 'Maça',
    __v: 0,
    _id: '5d08c85f4d6ea30b43bf6d3b'
  },
  {
    data: '2019-11-08T22:16:00.000Z',
    guarnicao1: 'Batata Frita',
    guarnicao2: 'Ameixa da jamaica',
    opcao1: 'Omelete',
    opcao2: 'Bife',
    pratoPrincipal: 'Bacon',
    salada: 'Alface e tomate',
    sobremesa: 'Maça',
    __v: 0,
    _id: '5d08c85f4d6ea30b43bf6d3b'
  },
]