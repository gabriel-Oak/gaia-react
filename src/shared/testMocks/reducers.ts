import Action from "../../shared/interfaces/Action";
import { MainActionsTypes } from "../../pages/main/mainActionsTypes";

const MAIN_STATE = {
    title: 'Gaia',
    drawer: false,
    snackbar: {
        open: false,
    },
    user: {
        admin: true,
        email: 'test@email.cone',
        nome: 'Jorginho',
        user: 'string'
    },
}

export const mainReducerMock = (state = MAIN_STATE, action: Action) => {
    switch (action.type) {
        case MainActionsTypes.FETCH_USER:
            return { ...state, auth: true, user: action.value }
        default:
            return state;
    }
}
