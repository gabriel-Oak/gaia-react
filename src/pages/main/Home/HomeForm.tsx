import React from 'react'
import WeekTabs from '../../../shared/components/WeekTabs/WeekTabs';
import { reduxForm } from 'redux-form';
import HomeFormTab from './homeFormTab';
import { Button, Card, CardActions, CircularProgress } from '@material-ui/core';
import Menu from '../../../shared/interfaces/Menu';
import { connect } from 'react-redux';
import { ReducersPool } from '../../../reducers';

let HomeForm: any = (props: any) => {
  const { handleSubmit, onSubmit, tabIndex, setTab, loading, menus } = props;

  return (
    <form onSubmit={handleSubmit(onSubmit)}>
      {
        loading && !menus.length ?

          <div className="Center-container">
            <CircularProgress size={80} />
          </div> :

          <Card>

            <WeekTabs index={tabIndex} onChange={setTab}>

              {
                menus.map((menu: Menu, index: number) => (
                  <HomeFormTab
                    name={`pratoPrincipal${index}`}
                    menu={menu}
                    loading={loading}
                    key={index}
                  />
                ))
              }

            </WeekTabs>

            <CardActions className="button-container">
              <Button
                type="submit"
                variant="contained"
                color="primary"
                disabled={loading}
              >
                Trocar
              </Button>
            </CardActions>

          </Card>
      }
    </form>
  );
}

const mapStateToProps = (state: ReducersPool) => {
  const { homeReducer: { initialFormValues } } = state;

  return ({
    initialValues: initialFormValues
  });
};

HomeForm = reduxForm({ form: 'homeForm' })(HomeForm);

HomeForm = connect(mapStateToProps)(HomeForm);

export default HomeForm;
