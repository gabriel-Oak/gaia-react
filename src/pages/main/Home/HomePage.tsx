import React, { PureComponent, ReactNode } from 'react'

import { connect } from 'react-redux';
import { ReducersPool } from '../../../reducers';
import { setTitle } from '../mainActions';
import { setTab, fetchMenus } from './homeActions';
import { homeState } from './homeReducer';
import HomeForm from './HomeForm';
import { getSession } from '../../../shared/utils/auth';

import './Home.css';

interface Props extends homeState {
  setTitle: Function;
  setTab: Function;
  fetchMenus: Function;
  store?: any;
}

class HomePage extends PureComponent<Props> {

  constructor(props: Props) {
    super(props);
    this.props.setTitle('Home');
    
    this.props.fetchMenus(getSession().token)
    this.onSubmit = this.onSubmit.bind(this);
  }

  onSubmit(event: any) {
    console.log(event);
  }

  render(): ReactNode {

    return (
      <main className="Home">
        <article className="Center-container-horizontal s800">
        
          <HomeForm
            {...this.props}
            onSubmit={this.onSubmit}
          />

        </article>
      </main>
    )
  }
}

const mapStateToProps = (state: ReducersPool) => {
  const { homeReducer } = state;
  return homeReducer;
};

const mapDispatchToProps = {
  setTitle,
  setTab,
  fetchMenus
};


export default connect(mapStateToProps, mapDispatchToProps)(HomePage);
