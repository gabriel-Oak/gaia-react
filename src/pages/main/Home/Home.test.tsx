import React from 'react';
import ReactDOM from 'react-dom';

import { shallow } from 'enzyme';

import { createStore, applyMiddleware } from 'redux';
import reducers from '../../../reducers';
import thunk from 'redux-thunk';
import HomePage from './HomePage';
import HomeForm from './HomeForm';
import { menusMock } from '../../../shared/testMocks/menus';
import { Provider } from 'react-redux';

describe('NotFound', () => {
  let store: any;
  let page: any;

  const onSubmit = jest.fn();

  beforeEach(() => {
    store = createStore(
      reducers,
      applyMiddleware(thunk)
    );

    page = shallow(
      <HomePage store={store} />
    ).childAt(0).dive();
  });

  it('Should match snapshot', () => {
    expect(page).toMatchSnapshot();
  });

  describe('Home form', () => {

    it('should match snapshot', () => {
      const form = shallow(
        <HomeForm
          onSubmit={onSubmit}
          menus={menusMock}
          store={store}
        />
      ).dive().dive().dive().dive();

      expect(form).toMatchSnapshot();
    });

    it('should render the form without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(
        <Provider store={store}>
          <HomeForm
            onSubmit={onSubmit}
            menus={menusMock}
            store={store}
          />
        </Provider>
        , div
      );
      ReactDOM.unmountComponentAtNode(div);
    });
  });
});