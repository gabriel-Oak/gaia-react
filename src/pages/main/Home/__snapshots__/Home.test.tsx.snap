// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound Home form should match snapshot 1`] = `
<ReduxForm
  _reduxForm={null}
  destroyOnUnmount={true}
  dispatch={[Function]}
  enableReinitialize={false}
  forceUnregisterOnUnmount={false}
  form="homeForm"
  getFormState={[Function]}
  keepDirtyOnReinitialize={false}
  menus={
    Array [
      Object {
        "__v": 0,
        "_id": "5d08c85f4d6ea30b43bf6d3b",
        "data": "2019-11-04T22:16:00.000Z",
        "guarnicao1": "Batata Frita",
        "guarnicao2": "Ameixa da jamaica",
        "opcao1": "Omelete",
        "opcao2": "Bife",
        "pratoPrincipal": "Bacon",
        "salada": "Alface e tomate",
        "sobremesa": "Maça",
      },
      Object {
        "__v": 0,
        "_id": "5d08c85f4d6ea30b43bf6d3b",
        "data": "2019-11-05T22:16:00.000Z",
        "guarnicao1": "Batata Frita",
        "guarnicao2": "Ameixa da jamaica",
        "opcao1": "Omelete",
        "opcao2": "Bife",
        "pratoPrincipal": "Bacon",
        "salada": "Alface e tomate",
        "sobremesa": "Maça",
      },
      Object {
        "__v": 0,
        "_id": "5d08c85f4d6ea30b43bf6d3b",
        "data": "2019-11-06T22:16:00.000Z",
        "guarnicao1": "Batata Frita",
        "guarnicao2": "Ameixa da jamaica",
        "opcao1": "Omelete",
        "opcao2": "Bife",
        "pratoPrincipal": "Bacon",
        "salada": "Alface e tomate",
        "sobremesa": "Maça",
      },
      Object {
        "__v": 0,
        "_id": "5d08c85f4d6ea30b43bf6d3b",
        "data": "2019-11-07T22:16:00.000Z",
        "guarnicao1": "Batata Frita",
        "guarnicao2": "Ameixa da jamaica",
        "opcao1": "Omelete",
        "opcao2": "Bife",
        "pratoPrincipal": "Bacon",
        "salada": "Alface e tomate",
        "sobremesa": "Maça",
      },
      Object {
        "__v": 0,
        "_id": "5d08c85f4d6ea30b43bf6d3b",
        "data": "2019-11-08T22:16:00.000Z",
        "guarnicao1": "Batata Frita",
        "guarnicao2": "Ameixa da jamaica",
        "opcao1": "Omelete",
        "opcao2": "Bife",
        "pratoPrincipal": "Bacon",
        "salada": "Alface e tomate",
        "sobremesa": "Maça",
      },
    ]
  }
  onSubmit={[MockFunction]}
  persistentSubmitErrors={false}
  pure={true}
  shouldAsyncValidate={[Function]}
  shouldError={[Function]}
  shouldValidate={[Function]}
  shouldWarn={[Function]}
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
  submitAsSideEffect={false}
  touchOnBlur={true}
  touchOnChange={false}
  updateUnregisteredFields={false}
/>
`;

exports[`NotFound Should match snapshot 1`] = `
<main
  className="Home"
>
  <article
    className="Center-container-horizontal s800"
  >
    <Connect(ReduxForm)
      fetchMenus={[Function]}
      menus={Array []}
      onSubmit={[Function]}
      setTab={[Function]}
      setTitle={[Function]}
      store={
        Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        }
      }
      tabIndex={0}
    />
  </article>
</main>
`;
